src/management-system-v2/lib/sharing/process-sharing.ts

```diff
@@ -0,0 +1,51 @@
+'use server';
+
+import jwt from 'jsonwebtoken';
+import { getProcessMetaObjects, updateProcessMetaData } from '@/lib/data/legacy/_process';
+import { getCurrentEnvironment, getCurrentUser } from '@/components/auth';
+import { UserErrorType, userError } from '../user-error';
+import { toCaslResource } from '../ability/caslAbility';
+
+export interface TokenPayload {
+  processId: string | string[];
+  embeddedMode?: boolean;
+  timestamp?: number;
+}
+
+export interface ProcessGuestAccessRights {
+  shared?: boolean;
+  sharedAs?: 'public' | 'protected';
+  shareTimeStamp?: number;
+}
+
+export async function updateProcessGuestAccessRights(
```

Comment:
This function does basically the same as updateProcess in lib/data/processes.tsx